tests/test_routes.py

```diff
@@ -0,0 +1,211 @@
+def test_get_all_planets_with_no_records(client):
+    # Act
+    response = client.get("/planets")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert response_body == []
+    
+def test_get_all_planets(client, two_planets):
+    response = client.get('/planets')
+
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == [{"id":1,
+        "name": "Earth",
+        "description": "home",
+        "moons": 1,
+        "life": True},
+        {"id":2,
+        "name": "Mars",
+        "description": "1st Colony",
+        "moons": 0,
+        "life": True}]
+
+def test_get_planet_by_id(client, two_planets):
+    response = client.get("planets/1")
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == {
+        "id":1,
+        "name": "Earth",
+        "description": "home",
+        "moons": 1,
+        "life": True
+    }
+
+def test_get_planet_by_name(client, two_planets):
+    response = client.get("planets?name=Earth")
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == [{
+        "id":1,
+        "name": "Earth",
+        "description": "home",
+        "moons": 1,
+        "life": True
+    }]
+
+def test_get_planet_by_invalid_id(client):
+    response = client.get("planets/a")
+    response_body = response.get_json()
+
+    assert response.status_code == 400
+    assert response_body == {"details":"Invalid id a"}
+
+def test_create_one_planet(client):
+
+    response = client.post("/planets", json={
+        "name": "Pluto",
+        "description": "A real planet!",
+        "moons": 0,
+        "life": False 
+    })
+    response_body = response.get_json()
+
+    assert response.status_code == 201
+    assert response_body == {
+        "id":1,
```

Comment:
Here too we might want to avoid assuming the id of the created resource. We could check for the keys we are certain of (often, we write helper test methods to be able to check for subsets of keys/values), and ensure the type of the id is something we would expect.